api/nodes/permissions.py

```diff
@@ -134,6 +166,29 @@ def has_object_permission(self, request, view, obj):
             return node.has_permission(auth.user, osf_permissions.ADMIN)
 
 
+class NodeGroupDetailPermissions(permissions.BasePermission):
+    """Permissions for node group detail - involving who can update the relationship
+    between a node and an OSF Group."""
+
+    acceptable_models = (OSFGroup, AbstractNode,)
+
+    def load_resource(self, context, view):
+        return AbstractNode.load(context[view.node_lookup_url_kwarg])
+
+    def has_object_permission(self, request, view, obj):
+        assert_resource_type(obj, self.acceptable_models)
+        auth = get_user_auth(request)
+        node = self.load_resource(request.parser_context['kwargs'], view)
+        if request.method in permissions.SAFE_METHODS:
+            return node.is_public or node.can_view(auth)
+        elif request.method == 'DELETE':
+            # If deleting an OSF group from a node, you either need admin perms
+            # or you need to be an OSF group manager
+            return node.has_permission(auth.user, osf_permissions.ADMIN) or obj.has_permission(auth.user, 'manage')
+        else:
+            return node.has_permission(auth.user, osf_permissions.ADMIN)
```

Comment:
idk, I value consistency a lot in these patterns we use often.  We're often doing something like this, 1)is it a safe method 2) Is it a delete method? 3) Otherwise do this for all other methods.  Every element of the conditional is related to the method.  

I personally find this less readable: 1) are you an admin? 2) is this a delete method ...and so on...they are different questions.